RecoParticleFlow/PFClusterProducer/src/PFClusterEMEnergyCorrector.cc

```diff
@@ -14,81 +11,108 @@ namespace {
-    
-    _condnames_sigma_25ns.push_back("ecalPFClusterCorV2_EB_pfSize1_sigma_25ns");
-    _condnames_sigma_25ns.push_back("ecalPFClusterCorV2_EB_pfSize2_sigma_25ns");
-    _condnames_sigma_25ns.push_back("ecalPFClusterCorV2_EB_pfSize3_ptbin1_sigma_25ns");
-    _condnames_sigma_25ns.push_back("ecalPFClusterCorV2_EB_pfSize3_ptbin2_sigma_25ns");
-    _condnames_sigma_25ns.push_back("ecalPFClusterCorV2_EB_pfSize3_ptbin3_sigma_25ns");
-    _condnames_sigma_25ns.push_back("ecalPFClusterCorV2_EE_pfSize1_sigma_25ns");
-    _condnames_sigma_25ns.push_back("ecalPFClusterCorV2_EE_pfSize2_sigma_25ns");
-    _condnames_sigma_25ns.push_back("ecalPFClusterCorV2_EE_pfSize3_ptbin1_sigma_25ns");
-    _condnames_sigma_25ns.push_back("ecalPFClusterCorV2_EE_pfSize3_ptbin2_sigma_25ns");
-    _condnames_sigma_25ns.push_back("ecalPFClusterCorV2_EE_pfSize3_ptbin3_sigma_25ns");
-  }
-  
+   srfAwareCorrection_ = conf.getParameter<bool>("srfAwareCorrection");
+   
+   if (srfAwareCorrection_) {
+     condnames_mean_.push_back("ecalPFClusterCor2017V2_EB_ZS_mean_25ns");
+     condnames_mean_.push_back("ecalPFClusterCor2017V2_EB_Full_ptbin1_mean_25ns");
+     condnames_mean_.push_back("ecalPFClusterCor2017V2_EB_Full_ptbin2_mean_25ns");
+     condnames_mean_.push_back("ecalPFClusterCor2017V2_EB_Full_ptbin3_mean_25ns");
+     condnames_mean_.push_back("ecalPFClusterCor2017V2_EE_ZS_mean_25ns");
+     condnames_mean_.push_back("ecalPFClusterCor2017V2_EE_Full_ptbin1_mean_25ns");
+     condnames_mean_.push_back("ecalPFClusterCor2017V2_EE_Full_ptbin2_mean_25ns");
+     condnames_mean_.push_back("ecalPFClusterCor2017V2_EE_Full_ptbin3_mean_25ns");
+     
+     condnames_sigma_.push_back("ecalPFClusterCor2017V2_EB_ZS_sigma_25ns");
+     condnames_sigma_.push_back("ecalPFClusterCor2017V2_EB_Full_ptbin1_sigma_25ns");
+     condnames_sigma_.push_back("ecalPFClusterCor2017V2_EB_Full_ptbin2_sigma_25ns");
+     condnames_sigma_.push_back("ecalPFClusterCor2017V2_EB_Full_ptbin3_sigma_25ns");
+     condnames_sigma_.push_back("ecalPFClusterCor2017V2_EE_ZS_sigma_25ns");
+     condnames_sigma_.push_back("ecalPFClusterCor2017V2_EE_Full_ptbin1_sigma_25ns");
+     condnames_sigma_.push_back("ecalPFClusterCor2017V2_EE_Full_ptbin2_sigma_25ns");
+     condnames_sigma_.push_back("ecalPFClusterCor2017V2_EE_Full_ptbin3_sigma_25ns");
+   } else {
+     if (applyMVACorrections_) {
+       autoDetectBunchSpacing_ = conf.getParameter<bool>("autoDetectBunchSpacing");
+       
+       if (autoDetectBunchSpacing_) {
+	 bunchSpacing_ = cc.consumes<unsigned int>(edm::InputTag("bunchSpacingProducer"));
+	 bunchSpacingManual_ = 0;
+       }
+       else {
+	 bunchSpacingManual_ = conf.getParameter<int>("bunchSpacing");
+       }
 
-  
-}
+       condnames_mean_25ns_.push_back("ecalPFClusterCorV2_EB_pfSize1_mean_25ns");
+       condnames_mean_25ns_.push_back("ecalPFClusterCorV2_EB_pfSize2_mean_25ns");
+       condnames_mean_25ns_.push_back("ecalPFClusterCorV2_EB_pfSize3_ptbin1_mean_25ns");
```

Comment:
for continuity in comparisons and commit flow it would be much better to start from the original version and then add the new piece for srfAwareCorrection_. 
Current version starts by removing all of the old, and then on top of the new code adds back in no particular order the old code.